### PR TITLE
Update snapshot APIs and add validateSnapshot

### PR DESCRIPTION
**Pull request by shilgapira**
## Related PRs
https://github.com/descope/go-sdk/pull/416

## Description
- Update API and documentation to use snapshots
- Update README

## Must
- [X] Tests
- [X] Documentation (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -808,16 +808,32 @@ With using a company management key you can get a list of all the projects in th
...
 ```typescript
 // Exports the current state of the project
-const files = await descopeClient.management.project.export();
+const exportRes = await descopeClient.management.project.exportSnapshot();
+```
+
+You can also import previously exported snapshots into the same project or a different one:
+
+```typescript
+const validateReq: ValidateSnapshotRequest = {
```
Comment (asafshen) [inline README.md:821]: I don't think that this type is exported, you exported it from types file, but not from the index file
this typing things is a big thing, we can probably discuss this for days about how to achieve that, but we are not there yet :) for now lets just remove the type from doc

File: lib/management/types.ts [hunk 2]
```diff
@@ -391,6 +393,17 @@ export type UserPasswordDjango = {
   hash: string; // the django hash in plaintext format, for example "pbkdf2_sha256$..."
 };
 
+export type UserPasswordPhpass = {
```
Comment (asafshen) [inline lib/management/types.ts:396]: following my comment on the readme I think this is the only type that is exported, see for example the user mapping

image

image

also, specifically for this `UserPasswordPhpass` and `UserPasswordMd5`, are they used anywhere?

**Review by asafshen — COMMENTED** [review]
cool, just some comments about typings, lmk what you think

File: README.md (hunk 1 above)
Comment (shilgapira) [reply README.md]: I've updated the README. Do we usually export misc types like these? Or is it enough for normal use that they are defined and then available during coding via autocomplete?

File: lib/management/types.ts (hunk 2 above)
Comment (shilgapira) [reply lib/management/types.ts:396]: yeah, they're used indirectly in `inviteBatch`

File: README.md (hunk 1 above)
Comment (asafshen) [reply README.md]: I'm not sure how / where it is used , usually such data is defined on the return value , and we don't export most of it 

Also note that if you don't export from index.ts then it's not exported 

I would remove it for now

File: lib/management/types.ts (hunk 2 above)
Comment (asafshen) [reply lib/management/types.ts:396]: What do you mean by that?

**Review by asafshen — APPROVED** [review]

**Pull request merged**